File: Dddml.Wms.Common/Domain/InventoryPostingRule/InventoryPostingRuleIds.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Dddml.Wms.Domain.InventoryPostingRule
{
    public static class InventoryPostingRuleIds
    {
        public const string QuantityOnHandUpdateProductSellableTotal = "OH->PrdSellableTotal";
        public const string QuantityOccupiedUpdateProductSellableTotal = "Oc->PrdSellableTotal";
        public const string QuantityReservedUpdateProductSellableTotal = "R->PrdSellableTotal";
        public const string QuantityVirtualUpdateProductSellableTotal = "V->PrdSellableTotal";
        public const string QuantityInTransitUpdateProductSellableTotal = "IT->PrdSellableTotal";

        // 需求数量 = 订单占用数量（Oc） - 在库数量（OH） + 保留数量（R） - 在途数量（IT）
        public const string QuantityOnHandUpdateProductRequiredTotal = "OH->PrdRequiredTotal";
        public const string QuantityOccupiedUpdateProductRequiredTotal = "Oc->PrdRequiredTotal";
        public const string QuantityReservedUpdateProductRequiredTotal = "R->PrdRequiredTotal";
        public const string QuantityInTransitUpdateProductRequiredTotal = "IT->PrdRequiredTotal";

        public const string OutputAccountNameSellableQuantity = "SellableQuantity";
        public const string OutputAccountNameRequiredQuantity = "RequiredQuantity";

    }
}

File: java/Dddml.Wms.JavaServices/src/main/java/org/dddml/wms/domain/inventorypostingrule/InventoryPostingRuleIds.java
package org.dddml.wms.domain.inventorypostingrule;

/**
 * Created by yangjiefeng on 2018/1/29.
 */
public class InventoryPostingRuleIds {

    public static final String QUANTITY_ON_HAND_UPDATE_PRODUCT_SELLABLE_TOTAL = "OH->PrdSellableTotal";
    public static final String QUANTITY_OCCUPIED_UPDATE_PRODUCT_SELLABLE_TOTAL = "Oc->PrdSellableTotal";
    public static final String QUANTITY_RESERVED_UPDATE_PRODUCT_SELLABLE_TOTAL = "R->PrdSellableTotal";
    public static final String QUANTITY_VIRTUAL_UPDATE_PRODUCT_SELLABLE_TOTAL = "V->PrdSellableTotal";
    public static final String QUANTITY_IN_TRANSIT_UPDATE_PRODUCT_SELLABLE_TOTAL = "IT->PrdSellableTotal";

    // 需求数量 = 订单占用数量（Oc） - 在库数量（OH） + 保留数量（R） - 在途数量（IT）
    public static final String QUANTITY_ON_HAND_UPDATE_PRODUCT_REQUIRED_TOTAL = "OH->PrdRequiredTotal";
    public static final String QUANTITY_OCCUPIED_UPDATE_PRODUCT_REQUIRED_TOTAL = "Oc->PrdRequiredTotal";
    public static final String QUANTITY_RESERVED_UPDATE_PRODUCT_REQUIRED_TOTAL = "R->PrdRequiredTotal";
    public static final String QUANTITY_IN_TRANSIT_UPDATE_PRODUCT_REQUIRED_TOTAL = "IT->PrdRequiredTotal";

    public static final String OUTPUT_ACCOUNT_NAME_SELLABLE_QUANTITY = "SellableQuantity";
    public static final String OUTPUT_ACCOUNT_NAME_REQUIRED_QUANTITY = "RequiredQuantity";
    
}
